src/main/java/slimeknights/tconstruct/gadgets/item/slimesling/MagmaSlimeSlingItem.java

```diff
@@ -0,0 +1,79 @@
+package slimeknights.tconstruct.gadgets.item.slimesling;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.util.math.RayTraceContext;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.world.World;
+import slimeknights.tconstruct.common.Sounds;
+import slimeknights.tconstruct.library.network.TinkerNetwork;
+import slimeknights.tconstruct.tools.common.network.EntityMovementChangePacket;
+
+import java.util.List;
+
+public class MagmaSlimeSlingItem extends BaseSlimeSlingItem {
+
+  public MagmaSlimeSlingItem(Properties props) {
+    super(props);
+  }
+
+  /** Called when the player stops using an Item (stops holding the right mouse button). */
+  @Override
+  public void onPlayerStoppedUsing(ItemStack stack, World worldIn, LivingEntity entityLiving, int timeLeft) {
+    if (!(entityLiving instanceof PlayerEntity)) {
+      return;
+    }
+
+    PlayerEntity player = (PlayerEntity) entityLiving;
+    float f = getForce(stack, timeLeft);
+
+    float range = 5F;
+    Vector3d start = player.getEyePosition(1F);
+    Vector3d look = player.getLookVec();
+    Vector3d direction = start.add(look.x * range, look.y * range, look.z * range);
+    AxisAlignedBB bb = player.getBoundingBox().expand(look.x * range, look.y * range, look.z * range).expand(1, 1, 1);
+    double minX = Math.min(start.x, direction.x);
+    double maxX = Math.max(start.x, direction.x);
+    double minY = Math.min(start.y, direction.y);
+    double maxY = Math.max(start.y, direction.y);
+    double minZ = Math.min(start.z, direction.z);
+    double maxZ = Math.max(start.z, direction.z);
+    List<Entity> entitiesInArea = player.getEntityWorld().getEntitiesWithinAABBExcludingEntity(player, bb);
+    double dist = range;
+    Entity closestEntity = null;
+    for (Entity entity : entitiesInArea) {
+      if (entity.getBoundingBox().intersects(minX, minY, minZ, maxX, maxY, maxZ)) {
+        if (look.distanceTo(entity.getLookVec()) < dist) {
+          dist = look.distanceTo(entity.getLookVec());
+          closestEntity = entity;
+        }
+      }
+    }
+
+    // cancel if there's a block in the way
+    BlockRayTraceResult mop = rayTrace(worldIn, player, RayTraceContext.FluidMode.NONE);
+    if (mop.getType() == RayTraceResult.Type.BLOCK && dist * dist < mop.getPos().distanceSq(player.getPosX(), player.getPosY(), player.getPosZ(), true)) {
+      return;
+    }
+
+    if (closestEntity != null) {
+      if (closestEntity instanceof LivingEntity) {
+        ((LivingEntity) closestEntity).applyKnockback(f , -look.x, -look.z);
```

Comment:
I am wondering if this is a bit too strong. Currently at max value this goes up to 6.0. For comparison, a level of knockback is 0.5, so effectively this is knockback 12. Maybe we should half it so its just knockback 6?